src/errorline.jl

```diff
@@ -0,0 +1,179 @@
+@userplot ErrorLine
+""" 
+# StatsPlots.errorline(x, y, arg): 
+    Function for parsing inputs to easily make a ribbons (https://ggplot2.tidyverse.org/reference/geom_ribbon.html) 
+    or errorbar (https://www.mathworks.com/help/matlab/ref/errorbar.html) plot while allowing for easily controlling error
+    type and NaN handling.
+
+# Inputs: default values are indicated with *s
+
+    x (vector, unit range) - the values along the x-axis for each y-point
+
+    y (matrix [x, repeat, group]) - values along y-axis wrt x. The first dimension must be of equal length to that of x.
+        The second dimension is treated as the repeated observations and error is computed along this dimension. If the 
+        matrix has a 3rd dimension this is treated as a new group.
+
+    ErrorStyle (symbol - *:Ribbon* or :Classic) - determines whether to use a ribbon style or stick style error representation.
+
+    CenterType (symbol - *:Mean* or :Median) - which approach to use to represent the central value of y at each x-value.
+
+    ErrorType (symbol - *:STD*, :SEM, :Percentile) - which error metric to use to show the distribution of y at each x-value.
+
+    Percentiles (Vector{Int64} *[25, 75]*) - if using ErrorType == :Percentile then which percentiles to use as bounds.
+
+    GroupColor (Symbol, RGB, Vector of Symbol or RGB) - Declares the color for each group. If no value is passed then will use
+        the default colorscheme. If one value is given then it will use that color for all groups. If multiple colors are 
+        given then it will use a different color for each group.
+
+    StickColor (Symbol, RGB, :Matched - *:Gray60*) - When using classic mode this will allow for the setting of the stick color.
+        If ":Matched" is given then the color of the sticks with match that of the main line.
+
+    StickWidthMultiplier (Float64 *.01*) - How much of the x-axis the horizontal aspect of the error stick should take up.
+
+# Example
+    ```julia
+    x = 1:10
+    y = fill(NaN, 10, 100, 3)
+    for i = axes(y,3)
+        y[:,:,i] = collect(1:2:20) .+ rand(10,100).*5 .* collect(1:2:20) .+ rand()*100
+    end
+    errorline(1:10, y)
+    ```
+"""
+errorline
+
+function compute_error(y::AbstractMatrix, CenterType::Symbol, ErrorType::Symbol, Percentiles::Vector{Int64})
```

Comment:
```suggestion
function compute_error(y::AbstractMatrix, CenterType::Symbol, ErrorType::Symbol, Percentiles::Vector{Int})
```
This currently fails the tests on 32bit machines. Any reason this couldn't be `percentiles::AbstractVector`?